compute/data_spark_version_test.go

```diff
@@ -0,0 +1,190 @@
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"long_term_support": true,
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.NoError(t, err)
+	assert.Equal(t, "7.3.x-scala2.12", d.Id())
+}
+
+func TestSparkVersionGpuMl(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"gpu": true,
+			"ml":  true,
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.NoError(t, err)
+	assert.Equal(t, "7.1.x-gpu-ml-scala2.12", d.Id())
+}
+
+func TestSparkVersionGenomics(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"genomics": true,
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.NoError(t, err)
+	assert.Equal(t, "7.3.x-hls-scala2.12", d.Id())
+}
+
+func TestSparkVersionLight(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"light": true,
+			"scala": "2.11",
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.NoError(t, err)
+	assert.Equal(t, "apache-spark-2.4.x-scala2.11", d.Id())
+}
+
+func TestSparkVersion300(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"spark_version": "3.0.0",
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.NoError(t, err)
+	assert.Equal(t, "7.1.x-scala2.12", d.Id())
+}
+
+func TestSparkVersionBeta(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"beta": true,
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.NoError(t, err)
+	assert.Equal(t, "7.5.x-scala2.12", d.Id())
+}
+
+func TestSparkVersionErrorNoResults(t *testing.T) {
+	_, err := qa.ResourceFixture{
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"beta":              true,
+			"long_term_support": true,
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.Error(t, err)
+	assert.Equal(t, true, strings.Contains(err.Error(), "query returned no results"))
+}
+
+func TestSparkVersionErrorMultipleResults(t *testing.T) {
+	_, err := qa.ResourceFixture{
+		Fixtures:    commonFixtures(),
+		Read:        true,
+		Resource:    DataSourceSparkVersion(),
+		NonWritable: true,
+		State: map[string]interface{}{
+			"latest": false,
+		},
+		ID: ".",
+	}.Apply(t)
+	assert.Error(t, err)
+	assert.Equal(t, true, strings.Contains(err.Error(), "query returned multiple results"))
```

Comment:
shouldn't the query do latest by default?

Reply:
not in this case - it explicitly says `latest = false`, just to trigger error handling...